### PR TITLE
removed addon polling

### LINKED ISSUE #3264: Sort out the polling code troubles
I'm on the outside here so I'm trying to follow what happened, but I think:

- Some patches were written in #3007 that were cool but unrelated to the issue #2997 that the PR was for
- #3014 was filed to rebase and land those patches, and that was done in #3123 .  This patch landed and was pushed live.
- #3196 was filed that things were broken in production and #3197 was written and pushed live to revert #3123 

So, all the issues and PRs are closed, but we're back to the first bullet point up there.  I think we need a new issue (it can be this one) to remove the polling code again, only this time make sure it doesn't break anything. :)

### PR DESCRIPTION
**Pull request by dannycoates**
see #3123 
fixes #3264

### CODE REVIEW

**Comment by MozillaTestPilotStorybook** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/3425/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/3b00adeb6b9c091e1e3a78a7e79e2e85cc3366da/

**Comment by MozillaTestPilotStorybook** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/3425/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/e1e5418ae76a10a30c2e6636748437ab61da040f/

**Comment by MozillaTestPilotStorybook** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/3425/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/12b7f2936e5a13fa416921b6310a58f51aa1d7e1/

File: frontend/src/app/containers/App/index.js [hunk 1]
```diff
@@ -109,21 +105,6 @@ class App extends Component {
   }
 
   componentDidMount() {
-    const userAgent = navigator.userAgent.toLowerCase();
-    this.props.setHasAddon(!!window.navigator.testpilotAddon);
```
Comment (dannycoates) [inline frontend/src/app/containers/App/index.js:113]: I started to pull this thread... "Can I remove this line?" and wound up unraveling the rest of addon and browser initialization. It turns out `shouldShowUpgradeWarning` relies on a set of assumptions that came unglued when `hasAddon` doesn't start out as null. Anyway, the browser state is static once initialized, so I moved it into the initial state, which saves a lap around the event loop and also ensure the rest of the conditions for `shouldShowUpgradeWarning` are in sync.

File: frontend/src/app/reducers/browser.js [hunk 2]
```diff
@@ -1,7 +1,5 @@
-// @flow
```
Comment (dannycoates) [inline frontend/src/app/reducers/browser.js:1]: `navigator.mozAddonManager` below is breaking flow

File: frontend/src/app/lib/InstallManager.js [hunk 3]
```diff
@@ -169,51 +157,49 @@ export function setupAddonConnection(store) {
   mam.addEventListener('onPropertyChanged', (addon, p) => {
   });
 */
-  getExperimentAddons(store.getState().experiments.data).then(addons => {
-    const enabled = addons.filter(a => a && a.isEnabled);
-    const installed = {};
-    enabled.forEach(a => {
-      installed[a.id] = {
+  mam.getAddonByID(TESTPILOT_ADDON_ID).then(addon => {
+    store.dispatch(addonActions.setHasAddon(!!addon));
```
Comment (dannycoates) [inline frontend/src/app/lib/InstallManager.js:163]: this replaces the need for `navigator.testpilotAddon`... i think

File: frontend/src/app/lib/InstallManager.js [hunk 4]
```diff
@@ -86,73 +84,63 @@ export function uninstallAddon() {
...
-        const x = experiments.data[i];
-        store.dispatch(addonActions.enableExperiment(x));
-        store.dispatch(
-          updateExperiment(x.addon_id, {
-            inProgress: false,
-            error: false
-          })
-        );
-      }
+    if (!addon) { return false; }
```
Comment (dannycoates) [inline frontend/src/app/lib/InstallManager.js:98]: most all the changes in this file come straight from #3123

File: frontend/src/app/lib/InstallManager.js [hunk 5]
```diff
@@ -86,73 +84,63 @@ export function uninstallAddon() {
...
+    if (addon.id === TESTPILOT_ADDON_ID) {
+      return store.dispatch(addonActions.setHasAddon(false));
+    }
+    const { experiments } = store.getState();
```
Comment (meandavejustice) [inline frontend/src/app/lib/InstallManager.js]: This loop looks like a duplicate of line 97 above, can they be consolidated into a function?

**Review by meandavejustice — CHANGES_REQUESTED** [review]
This is my only comment for now

**Comment by MozillaTestPilotStorybook** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/3425/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/54dd68244589ecbd6cdaaebc211d2829ce375ba2/

**Comment by MozillaTestPilotStorybook** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/3425/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/6c0bd61be30e999933f3ecef21ee009a8af3542c/

**Comment by dannycoates** [comment]
@meandavejustice updated

**Comment by MozillaTestPilotStorybook** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/3425/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/c5cfb013ad0ca3beb2c5504e40107e4eb2102405/

**Pull request merged**